test/strategies/sonatype/index.unit.js

```diff
@@ -132,3 +132,28 @@ test("sonatype strategy: hydratePayloadDependencies when using NodeSecure standa
   restoreHttpAgent();
   tape.end();
 });
+
+test("sonatype strategy: fetchDataForPackageURLs with coordinates exceeding the ratelimit", async(tape) => {
+  const [mockedHttpAgent, restoreHttpAgent] = setupHttpAgentMock();
+  const mockedHttpClient = mockedHttpAgent.get(kSonatypeOrigin);
+
+  const coordinates = Array.from({ length: 200 }, () => kFakePackageURL);
+  mockedHttpClient
+    .intercept({
+      path: kSonatypeApiPath,
+      method: "POST"
+    })
+    .reply(200, [kSonatypeVulnComponent], kHttpClientHeaders)
+    .times(2);
+
+  // Note: pendingInterceptor.timesInvoked always equal O ?
+  const [pendingInterceptor] = mockedHttpAgent.pendingInterceptors();
+  const response = await fetchDataForPackageURLs(coordinates);
```

Comment:
In my opinion ideally we should not directly export and call `fetchDataForPackageURLs` which is an implementation detail for the Sonatype strategy. We could call `hydratePayloadDependencies` which is the public API and will internally call `fetchDataForPackageURLs`.

Nevertheless this entirely depends on the capabilities we have with the mocked agent to assert things directly without having directly the result of the responses. Can't we get back both request results in the same spirit as what you did using the mocked http agent? Can we also know how many times the agent was hit? If we can't know that, then we can stick with that approach, otherwise I'd suggest to stay far from internals of the strategy